test/infamy/iface.py

```diff
@@ -0,0 +1,54 @@
+def extract_iface_param(row, key):
+    interfaces = row.get('interfaces')
+    if interfaces:
+        interface_list = interfaces.get('interface')
+        if interface_list and isinstance(interface_list, list):
+            for interface in interface_list:
+                if key in interface:
+                    return interface[key]
+    return None 
```

Comment:
The staircase effect in this method is a good example of a structure I personally try to avoid.  Exit early and use if-statements to check for error cases, letting the main part of the function continue.

Example pseudo code:

```python
   if Not interfaces:
      return None

   for interface in interfaces.get('interface'):
      if key Not in interface:
         continue
      return interface[key]

   return None
```

Reply:
Latest changes look much better.  Great work on this, Ahmed!